node/pkg/driver/node.go

```diff
@@ -132,6 +133,10 @@ func (d *NodeService) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Wrong connectivity type %s", connectivityType))
 	}
 
+	if err := osDeviceConnectivity.EnsureLogin(ipsByArrayInitiator); err != nil {
```

Comment:
You should do it only once, it is a bad practice to do iscsi login in every nodeStage request.

Reply:
is it better to parse all sessions (`iscsiadm -m session`) in every nodeStage request?

Reply:
Yes, get current sessions and parse them to see if it is already login is better.
But you still need to run get_session in every request, is it fast enough that we can ignore its effect?

Reply:
if there are a lot of sessions (even sessions that are not related to us), it might take a lot of time (for every request)

Reply:
so you need find another way.

Reply:
I couldn't find another way. If you find another way, please write it here and I will consider it.
Until then, I will leave it as is.

Reply:
We currently have two options, for every request:
1. Login
or
2. Parse all sessions to avoid some (probably most) logins

Which do you prefer?

Reply:
2 for now. If we don't have a better way.

Reply:
Done